Autopilot/Test/Src/TelemetryManager/Test_TelemetryFSM.cpp

```diff
@@ -0,0 +1,110 @@
+#include <gtest/gtest.h>
+
+#include "telemetryManager.hpp"
+#include "telemetryStateClasses.hpp"
+
+
+using namespace std; 
+using ::testing::Test;
+
+TEST(TelemetryManagerFSM, InitialStateIsInitialMode){
+    /***SETUP***/
+    telemetryManager telemetryMng;
+    /***ASSERTS***/
+    ASSERT_EQ(*(telemetryMng.getCurrentState()), initialMode::getInstance());
+}
+
+TEST(TelemetryManagerFSM, InitialTransitionToGround){
+    /***SETUP***/
+    telemetryManager telemetryMng;
+    /***STEPTHROUGH***/
+    telemetryMng.execute();
+    /***ASSERTS***/
+    ASSERT_EQ(*(telemetryMng.getCurrentState()), groundDataPathMode::getInstance());
+}
+
+TEST(TelemetryManagerFSM, GroundToRegularThroughDataValid){
+    /***SETUP***/
+    telemetryManager telemetryMng;
+    /***DEPENDECIES***/
+    telemetryMng.dataValid=true;
+    telemetryMng.dataError=false;
+    /***STEPTHROUGH***/
+    telemetryMng.execute();
```

Comment:
Just curious, but why are you executing the telemetry manager state multiple times?

Reply:
I don't see the need of executing the telemetry manager twice